### PR TITLE
Replace postgress with mongo

### PR DESCRIPTION
**Pull request by knoll3**
View the readme for instructions on setting up the db. Still using docker compose as before but you need to set the env variable in `.env`. 
```
DATABASE_URL="mongodb://admin:pass@localhost:27017/prisma-mongo?authSource=admin&retryWrites=true&w=majority"
```

I set up a quick temporary test for the db in `scripts/index.ts`. This scripts directory can be removed, it's just there for quickly testing db insertions.
```
cd backend
npx ts-node scripts/index.ts
```

### CODE REVIEW

File: backend/prisma/schema.prisma [hunk 1]
```diff
@@ -6,30 +6,31 @@ generator client {
...
-  address   String
-  balance    Int
-  path      String[]
-  indices   String[]
-  root      String
-  createdAt DateTime  @default(now())
+model Leaf {
+  address     String   @map("_id")
+  usdValue    Int
+  merkleProof String[]
```
Comment (christopherdancy) [inline backend/prisma/schema.prisma:16]: Question regarding the merkle proof. Why did this change from path, indices, and root?

File: backend/prisma/schema.prisma [hunk 2]
```diff
@@ -6,30 +6,31 @@ generator client {
...
 }
 
-model Tree {
-  root        String
-  createdAt   DateTime  @default(now())
+model Group {
+  merkleRoot     String   @map("_id")
+  name           String
+  threshold      Int
+  erc20Addresses String[]
```
Comment (christopherdancy) [inline backend/prisma/schema.prisma:26]: I am unsure if we need to save erc20 addresses since they will be available from the leaf object

Comment (sethhrbek) [reply backend/prisma/schema.prisma:26]: @christopherdancy -- where are the ERC20addresses on the leaf object? The leaf is only the address, balance, and proof currently.

File: backend/prisma/schema.prisma (hunk 1 above)
Comment (sethhrbek) [reply backend/prisma/schema.prisma:16]: Yeah this should remain the same as path + indices --- and then we need to link the leaf to the group somehow (whether thats via a root other other)

**Pull request merged**